### PR TITLE
Fix: Gallery widget with dynamic tags can't be edited [ED-4443]

### LINKED ISSUE #15529: [🧐 Evaluating] 🐞 Bug Report: Gallery completely bugged when using ACF Dynamic Data as source
- [x] I have searched for similar issues in both open and closed tickets and cannot find a duplicate.
- [x] The issue still exists against the latest stable version of Elementor.

This looks like a rather large bug in the gallery widget - I'd love anyone else to confirm this happens to them too - I can consistently replicate it and it makes the gallery widget considerably less useful than it should be when you want to drive it from dynamic data

- Start a new template
- Drag a gallery widget out
- Add a couple of images to that gallery
- All works fine
- Now - you have to have an ACF gallery set up on the backend for this next part - mine is driven from an options page with the gallery field available
- Duplicate the gallery widget above
- Preview the page to verify it works
- Now change it from having static gallery images to being driven from the ACF Gallery field
- You can change the settings - it looks great
- Preview - this is where it starts to go wrong for me
- Refresh your editor page for the template or go back into it if you closed it
- Right-click the gallery element that is driven by dynamic data - the editor panel on the left is blank - this only happens if you have dynamic data - and you cannot change any settings
- Also via the navigator, you cannot delete any galleries that are set to be driven by dynamic data

This is replicable on a system with only Elementor, Elementor Pro, ACF Pro & Hello theme running

- [x] This bug happens with only Elementor plugin active (and Elementor Pro).
- [x] This bug happens with a default WordPress theme active.
- [x] I can reproduce this bug consistently using the steps above.

Screenshot 2021-07-01 at 11 21 13

Screenshot 2021-07-01 at 11 21 24

Screenshot 2021-07-01 at 11 21 34

### PR DESCRIPTION
**Pull request by StyleShit**
Resolves #15529

### CODE REVIEW

File: assets/dev/js/editor/container/container.js [hunk 1]
```diff
@@ -159,7 +159,7 @@ export default class Container extends ArgsObject {
 	}
 
 	initialize() {
-		if ( this.view ) {
+		if ( this.view && this.view instanceof Marionette.CompositeView ) {
```
Comment (matipojo) [inline assets/dev/js/editor/container/container.js:162]: Find a better solution

**Review by matipojo — CHANGES_REQUESTED** [review]

File: assets/dev/js/editor/container/container.js (hunk 1 above)
Comment (StyleShit) [reply assets/dev/js/editor/container/container.js]: Done.

File: assets/dev/js/editor/container/container.js [hunk 2]
```diff
@@ -464,6 +464,10 @@ export default class Container extends ArgsObject {
 		return Container.TYPE_REPEATER_ITEM === this.type;
 	}
 
+	isElement() {
+		return ! ! this.model.get( 'elType' );
+	}
+
```
Comment (iNewLegend) [inline assets/dev/js/editor/container/container.js:470]:
```suggestion
	isViewElement() {
		return this.view && this.model.get( 'elType' );
	}

```

File: assets/dev/js/editor/container/container.js [hunk 3]
```diff
@@ -159,7 +159,7 @@ export default class Container extends ArgsObject {
 	}
 
 	initialize() {
-		if ( this.view ) {
+		if ( this.view && this.isElement() ) {
```
Comment (iNewLegend) [inline assets/dev/js/editor/container/container.js:162]:
```suggestion
		if ( this.isViewElement() ) {
```

**Review by iNewLegend — CHANGES_REQUESTED** [review]

File: assets/dev/js/editor/container/container.js (hunk 3 above)
Comment (StyleShit) [reply assets/dev/js/editor/container/container.js:162]: Done.

File: assets/dev/js/editor/container/container.js (hunk 2 above)
Comment (StyleShit) [reply assets/dev/js/editor/container/container.js]: Done.

**Review by matipojo — APPROVED** [review]

**Pull request merged**